benchmarks/sindex.py

```diff
@@ -16,54 +20,82 @@ def generate_test_df():
         "points": points[points.is_valid],
         "polygons": polygons[polygons.is_valid],
     }
+    # ensure index is pre-generated
+    for data_type in data.keys():
+        data[data_type].sindex.query(data[data_type].geometry.values.data[0])
     return data
 
 
-class Bench:
+class BenchIntersection:
 
-    param_names = ["tree_geom_type"]
-    params = [["mixed", "points", "polygons"]]
+    param_names = ["input_geom_type", "tree_geom_type"]
+    params = [
+        geom_types,
+        geom_types,
+    ]
 
     def setup(self, *args):
         self.data = generate_test_df()
         # cache bounds so that bound creation is not counted in benchmarks
-        self.bounds = [g.bounds for g in self.data["mixed"].geometry]
+        self.bounds = {
+            data_type: [g.bounds for g in self.data[data_type].geometry]
+            for data_type in self.data.keys()
+        }
+
+    def time_intersects(self, input_geom_type, tree_geom_type):
+        tree = self.data[tree_geom_type].sindex
+        for bounds in self.bounds[input_geom_type]:
+            tree.intersection(bounds)
+
+
+class BenchIndexCreation:
+
+    param_names = ["tree_geom_type"]
+    params = [
+        geom_types,
+    ]
+
+    def setup(self, *args):
+        self.data = generate_test_df()
 
     def time_index_creation(self, tree_geom_type):
         """Time creation of spatial index.
 
-        Note: pygeos will only create the index once; this benchmark
-        is not intended to be used to compare rtree and pygeos.
+        Note: requires running a single query to ensure that
+        lazy-building indexes are actually built.
         """
-        self.data[tree_geom_type]._invalidate_sindex()
-        self.data[tree_geom_type]._generate_sindex()
-
-    def time_intersects(self, tree_geom_type):
-        for bounds in self.bounds:
-            self.data[tree_geom_type].sindex.intersection(bounds)
-
-    def time_intersects_objects(self, tree_geom_type):
-        for bounds in self.bounds:
-            self.data[tree_geom_type].sindex.intersection(bounds, objects=True)
+        self.data[tree_geom_type]._sindex_generated = None
```

Comment:
This attribute no longer exists?

Reply:
Yeah it's deleted but this is for this benchmark to work for both this branch and master. We could delete it right after this merge, but it might be nice to keep around if we ever want to do a test between a pre-this-PR state and post-this-PR state. I'll add a note about it.

Reply:
Added a note. Resolving for now.